### DISCUSSION TITLE
Accessing tabulator table?

### DISCUSSION THREAD

**Discussion by danbolinson-bt** [anvilistas/tabulator#68] [Q&A] [answered]
I think I'm missing something obvious...

Say I want to access a tabulator table method, like [```clearData```](http://tabulator.info/docs/4.0/update#alter-empty).

I'd expected to simply call ```self.tabulator_1.clearData()``` (where ```tabulator_1``` is the custom component added to my Form).

However, I get an ```AttributeError``` here. (I actually also get an attribute error when I try the ```previous_page``` as per the docs). 

So, how do I get the ```table``` object (like that referenced in the Tabulator docs)?

Thanks!

**Comment by s-cork** [comment, marked as answer]
That sounds odd. It should just work. Do you have a clone you can share?

**Reply by s-cork** [reply]
Just tried it on the clone you sent me the other day and `clearData()` works as expected. Maybe you changed the name of the tabulator component or something? Or you've pinned to a version other than 2.x?

**Reply by danbolinson-bt** [reply]
Sure, here is a clone link. I'm pinned to v2.1.5.

https://anvil.works/build#clone:VA3VTMAVOVTEGOI3=W2EKAGTD23TC2J3SRQBMEJUG

**Reply by s-cork** [reply]
Form show is too early. You should do this in the table_built event. 

See the readme section on [table built](https://github.com/s-cork/Tabulator#table-built).
